sdk/servicebus/Microsoft.Azure.WebJobs.Extensions.ServiceBus/src/Grpc/SettlementService.cs

```
@@ -0,0 +1,114 @@
+                }
+                return new SettlementReply();
+            }
+            return new SettlementReply {Exception = $"LockToken {request.Locktoken} not found."};
+        }
+
+        public override async Task<SettlementReply> Abandon(AbandonRequest request, ServerCallContext context)
+        {
+            if (_provider.ActionsCache.TryGetValue(request.Locktoken, out var action))
+            {
+                try
+                {
+                    await action.AbandonMessageAsync(
+                        action._eventArgs.Message,
+                        request.PropertiesToModify.ToDictionary(
+                            pair => pair.Key,
+                            pair => pair.Value.GetPropertyValue()),
+                        context.CancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception exception)
+                {
+                    return new SettlementReply() { Exception = exception.ToString() };
+                }
+                return new SettlementReply();
+            }
+            return new SettlementReply {Exception = $"LockToken {request.Locktoken} not found."};
+        }
+
+        public override async Task<SettlementReply> Defer(DeferRequest request, ServerCallContext context)
+        {
+            if (_provider.ActionsCache.TryGetValue(request.Locktoken, out var action))
+            {
+                try
+                {
+                    await action.DeferMessageAsync(
+                        action._eventArgs.Message,
+                        request.PropertiesToModify.ToDictionary(
+                            pair => pair.Key,
+                            pair => pair.Value.GetPropertyValue()),
+                        context.CancellationToken).ConfigureAwait(false);
+                }
+                catch (Exception exception)
+                {
+                    return new SettlementReply() { Exception = exception.ToString() };
```

You can let this bubble out and it will surface on the other side as an `RpcException` with Status of `INTERNAL`